src/Datadog.Trace.ClrProfiler.Native/IntegrationLoader.cpp

```diff
@@ -2,19 +2,34 @@
 
 using json = nlohmann::json;
 
-std::vector<integration> IntegrationLoader::load_integrations_from_file(const std::wstring file_path)
+std::vector<integration> IntegrationLoader::load_integrations_from_file(const std::wstring& file_path)
 {
     std::vector<integration> integrations;
 
     try
     {
-        json j;
+        std::ifstream stream;
+        stream.open(file_path);
+        integrations = load_integrations_from_stream(stream);
+        stream.close();
+    }
+    catch (...)
+    {
+        LOG_APPEND(L"failed to load integrations");
+    }
+
+    return integrations;
+}
+
+std::vector<integration> IntegrationLoader::load_integrations_from_stream(std::istream& stream)
+{
+    std::vector<integration> integrations;
 
-        // pipe in the file path
-        std::ifstream i;
-        i.open(file_path);
-        i >> j;
-        i.close();
+    try
+    {
+        json j;
+        // parse the stream
+        stream >> j;
 
         for (auto& el : j)
```

Comment:
If we pass in an object rather than a list, what happens? Does that cause the iteration to return type_error or does it still work? Consider adding a test for that case!